[maven/grpc] unable to find valid certification path to requested target

There's a runtime error in `grpc-example/grpc-example-java`. Using `mvn lagom:runAll` and then invoking `curl  http://localhost:9000/proxy/grpc-hello/Steve` causes an SSL handshake error (see stacktrace below).

The same `curl` operation works as expected when using the sbt build (`sbt runAll`) so this looks like a different dependency resolution or setting when using Maven.


```
18:11:45.834 [error] hello-proxy [] - Exception in RestCallId{method=GET, pathPattern='/proxy/grpc-hello/:id'}
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
Channel Pipeline: [SslHandler#0, ProtocolNegotiators$ClientTlsHandler#0, WriteBufferingAndExceptionHandler#0, DefaultChannelPipeline$TailContext#0]
	at io.grpc.Status.asRuntimeException(Status.java:533)
	at akka.grpc.internal.UnaryCallAdapter.onClose(UnaryCallAdapter.scala:40)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426)
	at io.grpc.internal.ClientCallImpl.access$500(ClientCallImpl.java:66)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:689)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$900(ClientCallImpl.java:577)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:751)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:740)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:326)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:269)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:264)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:645)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:464)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:360)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444)
	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1061)
	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1048)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:995)
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler.runAllDelegatedTasks(SslHandler.java:1494)
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler.access$1900(SslHandler.java:166)
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler$SslTasksRunner.run(SslHandler.java:1681)
	... 3 common frames omitted
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
	at java.base/sun.security.validator.Validator.validate(Validator.java:264)
	at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:276)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:141)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:623)
	... 14 common frames omitted
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
	at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
	... 20 common frames omitted
```

--- a/grpc-example/grpc-example-java/hello-impl/pom.xml
+++ b/grpc-example/grpc-example-java/hello-impl/pom.xml
@@ -44,13 +44,13 @@
         <dependency>
             <groupId>com.lightbend.play</groupId>
             <artifactId>play-grpc-runtime_${scala.binary.version}</artifactId>
-            <version>0.8.2</version>
+            <version>${play.grpc.version}</version>
         </dependency>
 
         <dependency>
             <groupId>com.lightbend.play</groupId>
             <artifactId>lagom-javadsl-grpc-testkit_${scala.binary.version}</artifactId>
-            <version>0.8.2</version>
+            <version>${play.grpc.version}</version>
             <scope>test</scope>
         </dependency>
 
@@ -105,6 +105,7 @@
                 <configuration>
                     <lagomService>true</lagomService>
                     <serviceHttpsPort>${hello-impl.https.port}</serviceHttpsPort>
+                    <serviceAddress>127.0.0.1</serviceAddress>
                     <serviceEnableSsl>true</serviceEnableSsl>
                 </configuration>
             </plugin>

--- a/grpc-example/grpc-example-java/hello-proxy-api/src/main/java/com/example/helloproxy/api/HelloProxyService.java
+++ b/grpc-example/grpc-example-java/hello-proxy-api/src/main/java/com/example/helloproxy/api/HelloProxyService.java
@@ -11,7 +11,6 @@ import static com.lightbend.lagom.javadsl.api.Service.*;
 public interface HelloProxyService extends Service {
 
     ServiceCall<NotUsed, String> proxyViaHttp(String id);
-
     ServiceCall<NotUsed, String> proxyViaGrpc(String id);
 
     default Descriptor descriptor() {

--- a/grpc-example/grpc-example-java/hello-proxy-impl/pom.xml
+++ b/grpc-example/grpc-example-java/hello-proxy-impl/pom.xml
@@ -46,7 +46,7 @@
         <dependency>
             <groupId>com.lightbend.play</groupId>
             <artifactId>play-grpc-runtime_${scala.binary.version}</artifactId>
-            <version>0.8.2</version>
+            <version>${play.grpc.version}</version>
         </dependency>
 
         <dependency>
@@ -69,6 +69,7 @@
                 <artifactId>lagom-maven-plugin</artifactId>
                 <configuration>
                     <lagomService>true</lagomService>
+                    <serviceEnableSsl>true</serviceEnableSsl>
                 </configuration>
             </plugin>
 

--- a/grpc-example/grpc-example-java/pom.xml
+++ b/grpc-example/grpc-example-java/pom.xml
@@ -77,11 +77,11 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <scala.binary.version>2.12</scala.binary.version>
         <lagom.version>1.6.3</lagom.version>
-        <play.version>2.8.1</play.version>
+        <play.version>2.8.2</play.version>
         <akka.http.version>10.1.12</akka.http.version>
         <akka.grpc.version>0.8.4</akka.grpc.version>
         <play.grpc.version>0.8.2</play.grpc.version>
 
-        <hello-impl.https.port>11011</hello-impl.https.port>
+        <hello-impl.https.port>11000</hello-impl.https.port>
     </properties>
 </project>

